src/Symfony/Component/Console/Application.php

```diff
@@ -923,6 +928,14 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
         $e = null;
 
         try {
+            if (class_exists(Lock::class) && \in_array(LockableTrait::class, class_uses($command))) {
+                $command->getDefinition()->addOption(new InputOption('--skip-lock', '', InputOption::VALUE_NONE, 'Skip checking for and setting a lock'));
+
+                if (false === $input->hasParameterOption(['--skip-lock'], true) && false === $this->lock($command->getName().'_symfony')) {
```

Comment:
the lock should be released() somewhere

Reply:
Agreed. I added a rough release mechanism for now. Though, I'm wondering if both of these parts can be moved into event listeners now that the methods are defined in an interface.